### ISSUE TITLE
"jsonc/no-comments" rule - [bug] some comments are ignored

### ISSUE THREAD

**Issue by rodrigojcmello** [ota-meshi/eslint-plugin-jsonc#239]
It is possible to simulate the error with the code:

```json
{
  /* hi */
  // test
  "name": "John"
}
```

or

```json
{
  // test
  "name": "John"
}
```

or

```json
{
  "name": "John",
  "test": "test",
  // test
  "test2": "test"
}
```

![plugin-jsonc](https://github.com/ota-meshi/eslint-plugin-jsonc/assets/5174526/43e75aa4-520b-4385-8c64-78a828720f66)

WebStorm has its own JSON validator, which means that WebStorm identifies the error because there should not be comments in a JSON file. However, the warning from eslint-plugin-jsonc does not appear. If you look at the terminal in the image, you will see the command to run eslint on the same file, and nothing happens because the error is not detected. Initially, I thought it was an error only with comments made at the beginning of the block, but that's not the case. I have managed to enforce the rule "jsonc/no-comments," but generally, it doesn't work.

**Comment by ota-meshi** [comment]
Do you have `{"rules":{"jsonc/no-comments":"error"}}` in your configuration? If not, please add it.
If so, please share your repository.

**Comment by rodrigojcmello** [comment]
Thank you for the response. Setting the rule directly worked... After some testing, I understood the problem. In the documentation example, 'plugin:jsonc/recommended-with-jsonc' is used. Since the library is called jsonc, I thought it would be the most obvious option to use to cover everything. However, further down it is mentioned that there are 3 recommended optional configurations, one for each type of JSON.

In other words, I misunderstood the documentation. So, to cover all 3 JSON formats, I need to do something like this:

```javascript
module.exports = {
  overrides: [
    {
      files: ['**/*.json'],

      extends: [
        'plugin:jsonc/recommended-with-json',
        'plugin:jsonc/prettier',
        'plugin:prettier/recommended',
      ],

      rules: {
        'prettier/prettier': [
          'error',
          {
            files: ['**/*.json'],
            parser: 'json',
          },
        ],
      },
    },
    {
      files: ['**/*.json5'],

      extends: [
        'plugin:jsonc/recommended-with-json5',
        'plugin:jsonc/prettier',
        'plugin:prettier/recommended',
      ],

      rules: {
        'prettier/prettier': [
          'error',
          {
            files: ['**/*.json5'],
            parser: 'json',
          },
        ],
      },
    },
    {
      files: ['**/*.jsonc'],

      extends: [
        'plugin:jsonc/recommended-with-jsonc',
        'plugin:jsonc/prettier',
        'plugin:prettier/recommended',
      ],

      rules: {
        'prettier/prettier': [
          'error',
          {
            files: ['**/*.jsonc'],
            parser: 'json',
          },
        ],
      },
    },
  ],
};

```
